### ISSUE TITLE
[Bug]: Backend name specified as module gets lowercased since 3.9

### ISSUE THREAD

**Issue by torokati44** [matplotlib/matplotlib#28432]
### Bug summary

`MPLBACKEND="module://foo.Bar"` now tries to import the `foo.bar` module as opposed to `foo.Bar`, as before.

### Code for reproduction

```Python
`MPLBACKEND="module://foo.Bar"`
```

### Actual outcome

`foo.bar` is imported

### Expected outcome

`foo.Bar` is imported

### Additional information

This probably regressed in https://github.com/matplotlib/matplotlib/pull/27948.

### Operating system

Fedora Linux 39

### Matplotlib Version

3.9.0

### Installation

None

**Comment by torokati44** [comment]
For context, this is a problem for us, because:
https://github.com/omnetpp/omnetpp/blob/master/ui/org.omnetpp.scave.pychart/python/omnetpp/internal/backend_SWTAgg.py
https://github.com/omnetpp/omnetpp/blob/0afc1f4e5cb1d2915c790f4b7790b4d23b34029c/ui/org.omnetpp.scave.pychart/src/org/omnetpp/scave/pychart/PythonProcessPool.java#L102

**Comment by anntzer** [comment]
I agree third-party backends should be matched case-sensitively (or really, matching python's import system behavior, perhaps https://peps.python.org/pep-0235/ (which may have become obsolete since it was written)).

**Comment by ianthomas23** [comment]
@torokati44 The change is intended. According to [PEP8](https://peps.python.org/pep-0008/#package-and-module-names) Python module names should be lowercase so your `backend_SWTAgg.py` should be `backend_swtagg.py`.

You can workaround this by adding a new `backend_swtagg.py` and importing (and hence effectively exporting) your `FigureCanvas`. But probably better is to rename your `backend_SWTAgg.py` to `backend_swtagg.py` and provide the import-export workaround in  `backend_SWTAgg.py` which you can deprecate and eventually remove.

**Comment by torokati44** [comment]
I see. A warning about it would have been nice about it though...
Anyway, we have already renamed the module internally as a workaround - it may end up as a permanent solution then.

**Comment by ksunden** [comment]
I'll add my opinion that I do not think we should have PEP8's style recommendations override the language implementation, forcing downstream projects to change capitalization. It may be best practice, but it is not universal practice.

Our own dependencies do not always have lower case names (looking at `PIL` and `IPython`) and while if it is only the last module name that is affected, that is workable. However, I would think this affects all parent modules, including the top level package names, which are much harder to change. 

Normalizing to lowercase for non `module://` backends I can absolutely get behind, but once you start getting to something that is directly imported, I think we should strive to handle all cases.

**Comment by ianthomas23** [comment]
If someone wishes to change the current behaviour they are welcome to do so. It is not something that I have any interest in doing.

**Comment by jklymak** [comment]
I think the point is that this is an unannounced regression.  Let's re-open to keep discussion going.

**Comment by greglucas** [comment]
I also think we should respect the casing of external user's requests. I think @ksunden's proposal about distinguishing between `module://` and built-ins is a good path forward.

> Normalizing to lowercase for non module:// backends I can absolutely get behind, but once you start getting to something that is directly imported, I think we should strive to handle all cases.

**Comment by torokati44** [comment]
Thanks for resolving this so quickly! :tada:
